spec/javascripts/MomentsSpec.js

```diff
@@ -1,105 +1,36 @@
-  var unselectAllAlliesWhoCanViewMoment;
 
+describe("Moments", function() {
   beforeAll(function() {
+    var newOrEdit;
     var elements = [];
-    elements.push("<div id='test_body'></div>");
-    elements.push("<label id='viewers_label'></label>");
+    elements.push("<div class='yes_title'></div>");
     elements.push("<input type='checkbox' id='viewers_all'></input>");
+    elements.push("<div id='viewers_list'><input type='checkbox' name='moment[viewers][]'></input></div>");
     for (var i = 0; i < elements.length; i++) {
       $(document.body).append(elements[i]);
     }
 
     newOrEdit = spyOn(window, 'newOrEdit');
-    isAllAlliesInputBoxIsChecked = spyOn(window, 'isAllAlliesInputBoxIsChecked');
-    selectAllAlliesWhoCanViewMoment = spyOn(window, 'selectAllAlliesWhoCanViewMoment');
-
-    setViewersCheckBoxToNotBeSelected = spyOn(window, 'setViewersCheckBoxToNotBeSelected');
-
-    unselectAllAlliesWhoCanViewMoment = spyOn(window, 'unselectAllAlliesWhoCanViewMoment');
-  });
-
-  it("has called newOrEdit when onReadyMoments is executed", function () {
-      onReadyMoments();
-
-      expect(newOrEdit).toHaveBeenCalled();
-  });
-
-  it("has set NO_ALLIES value to unselect all when creating or editing a new moment", function() {
-    newOrEdit.and.returnValue(true);
-
-    onReadyMoments();
-
-    expect(NO_ALLIES).toBe("Unselect all");
-  });
-
-  it("has set ALL_ALLIES value to select all when creating or editing a new moment", function() {
-    newOrEdit.and.returnValue(true);
-
-    onReadyMoments();
-
-    expect(ALL_ALLIES).toBe("Select all");
-  });
-
-  it("has set the default value of viewers_label text to ALL_ALLIES when creating or editing a new moment", function() {
     newOrEdit.and.returnValue(true);
 
-    expect($('#viewers_label').text()).toBe(ALL_ALLIES);
-  });
-
-  it("has called lableTextIsAllAllies when input box is checked", function() {
-      newOrEdit.and.returnValue(true);
-
-     $('#viewers_all').change();
-
-      expect(isAllAlliesInputBoxIsChecked).toHaveBeenCalled();
-  });
-
-  it("has selected all allies who can view moment when ALL_ALLIES is selected",  function() {
-      isAllAlliesInputBoxIsChecked.and.returnValue(true);
-
-      $('#viewers_all').change();
-
-      expect(selectAllAlliesWhoCanViewMoment).toHaveBeenCalled();
+    onReadyMomentsAndStrategies();
   });
 
-  it("has not selected #viewers_all checkbox when ALL_ALLIES is selected",  function() {
-      isAllAlliesInputBoxIsChecked.and.returnValue(true);
-
-      $('#viewers_all').change();
-
-      expect(setViewersCheckBoxToNotBeSelected).toHaveBeenCalled();
+  afterAll(function() {
+    $("#viewers_list").remove();
   });
 
-  it("has set #viewers_label text to NO_ALLIES when ALL_ALLIES is selected", function() {
-      isAllAlliesInputBoxIsChecked.and.returnValue(true);
+  it("has selected all allies who can view moment when \"Select all\" is selected",  function() {
+    $('#viewers_all').click();
```

Comment:
You can also test `expect($(":checkbox[name='moment[viewers][]']").eq(0).prop("checked")).toBe(false);` before clicking